### PR TITLE
Add TGB in GS

### PR DESCRIPTION
**Pull request by FlorianJacta**
Tries to solve: https://github.com/Avaiga/taipy/issues/526 by creating two ways to complete the GS. The change between the two is quite small.

The question is: is it confusing to have two ways to create the GS?

### CODE REVIEW

File: docs/getting_started/index.md [hunk 1]
```diff
@@ -116,62 +116,112 @@ While we've used Taipy's Python APIs to handle our scenario, it usually works se
...
+        ```python linenums="1"
+        import taipy as tp
+        import taipy.gui.builder as tgb
+
+        # Previous configuration of scenario
+        ...
+
+        with tgb.Page() as page:
+            tgb.text("Name:")
+            tgb.input("{input_name}")
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md:183]: I suspect that, when https://github.com/Avaiga/taipy/pull/635 is merged, that can be written:
`tgb.input(input_name)`

If I were you, I would give this a shot.

File: docs/getting_started/index.md [hunk 2]
```diff
@@ -116,62 +116,112 @@ While we've used Taipy's Python APIs to handle our scenario, it usually works se
...
-Name: <|{input_name}|input|>
-<|submit|button|on_action=submit_scenario|>
-Message: <|{message}|text|>
-"""
-```
 
-**Pages**
+        with tgb.Page() as page:
+            tgb.text("Name:")
+            tgb.input("{input_name}")
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md:214]: Same remark here (Simplified tgb syntax)

File: docs/getting_started/index.md [hunk 3]
```diff
@@ -116,62 +116,112 @@ While we've used Taipy's Python APIs to handle our scenario, it usually works se
...
-variable is a Markdown representation of the user interface. 
-It uses standard Markdown syntax as well as visual elements.
 
 **Visual elements**
 
 Taipy offers various visual elements that can interact with your Python variables and 
 environment. It allows you to display variables, modify them, and interact with the application.
 
-Here is a generic example of one: `<|{variable}|visual_element_type|...|>`. *variable* is 
+Here is a generic example of one: `<|{variable}|visual_element_type|...|>` with Markdown or `tgb.visual_element_type("{variable}", ...)` with the Python API. *variable* is
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: Same remark here (Simplified tgb syntax)
-> `tgb.visual_element_type(variable, ...)`

"with the Python API" -> "with the Page Builder API"

File: docs/getting_started/hello_world_tgb.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def submit_scenario(state):
+    state.scenario.input_name.write(state.input_name)
+    state.scenario.submit()
+    state.message = scenario.message.read()
+
+with tgb.Page() as page:
+    tgb.text("Name:")
+    tgb.input("{input_name}")
```
Comment (FabienLelaquais) [inline docs/getting_started/hello_world_tgb.py:37]: I suspect that, when https://github.com/Avaiga/taipy/pull/635 is merged, that can be written:
`tgb.input(input_name)`

If I were you, I would give this a shot.

**Review by FabienLelaquais — APPROVED** [review]
👌

File: docs/getting_started/index.md (hunk 1 above)
Comment (FlorianJacta) [reply docs/getting_started/index.md:183]: Wow, I didn't know about this one. This is impressive

File: docs/getting_started/index.md (hunk 3 above)
Comment (jrobinAV) [reply docs/getting_started/index.md]: Please break line before 100 chars.

File: docs/getting_started/index.md [hunk 5]
```diff
@@ -116,62 +116,112 @@ While we've used Taipy's Python APIs to handle our scenario, it usually works se
...
+        page = """
+        Name: <|{input_name}|input|>
+        <|submit|button|on_action=submit_scenario|>
+        Message: <|{message}|text|>
+        """
+        ```
+
+    === "Python"
+
+        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is built through the Taipy Gui Builder.
```
Comment (jrobinAV) [inline docs/getting_started/index.md]:
```suggestion
        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is built
        through the Taipy Gui Builder.
```

File: docs/getting_started/index.md [hunk 6]
```diff
@@ -116,62 +116,112 @@ While we've used Taipy's Python APIs to handle our scenario, it usually works se
...
+!!! example "GUI creation"
 
-# Previous configuration of scenario
-...
+    === "Markdown"
 
-page = """
-Name: <|{input_name}|input|>
-<|submit|button|on_action=submit_scenario|>
+        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is the Markdown representation of the page.
```
Comment (jrobinAV) [inline docs/getting_started/index.md:123]:
```suggestion
        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is the
        Markdown representation of the page.
```

**Review by jrobinAV — APPROVED** [review]
Looks good to me. Just minor remarks.

File: docs/getting_started/index.md [hunk 7]
```diff
@@ -168,7 +168,8 @@ Here's a simple GUI set up for our "Hello World" scenario:
 
     === "Python"
 
-        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is built through the Taipy Gui Builder.
+        Taipy pages can be defined in multiple ways: Markdown, Html or Python. *page* is built through
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md:171]: I don't think this line is useful, and the code does not define the tgb symbol.
If you really need to reference the taipy.gui.builder module, the proper wording is 'Page Builder API'. **Not GUI Builder**

File: docs/getting_started/index.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/getting_started/index.md:183]: That is for 3.1

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Do not use "Taipy GUI Builder". That will be confusing, with the Taipy GUI Builder coming up (Designer).
The proper wording is in the doc: Page Builder API - or "Python" for short.

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**